File: javascript-leetcode/283.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
  let lastNonZeroIdx = 0;
  for (let i = 0; i < nums.length; i += 1) {
    if (nums[i] !== 0) {
      nums[lastNonZeroIdx] = nums[i];
      lastNonZeroIdx += 1;
    }
  }
  for (let i = lastNonZeroIdx; i < nums.length; i += 1) {
    nums[i] = 0;
  }
};

File: golang-leetcode/283.go
package main

import "fmt"

func moveZeroes(nums []int) {
	j := 0
	for i := range nums {
		if nums[i] != 0 {
			nums[i], nums[j] = nums[j], nums[i]
			j++
		}
	}
}

func main() {
	fmt.Println("vim-go")
}
